### ISSUE TITLE
App create with HealthCheck get 422 when using port from ports array in HOST mode

### ISSUE THREAD

**Issue by max8899** [mesosphere/marathon#4282]
config:

``` json
{
    "backoffFactor": 1.5,
    "backoffSeconds": 1,
    "constraints": [
        [
            "allocation_group",
            "UNLIKE",
            "*"
        ]
    ],
    "container": {
        "docker": {
            "forcePullImage": true,
            "image": "nginx:1.9.11",
            "network": "HOST",
            "privileged": false
        },
        "type": "DOCKER"
    },
    "cpus": 0.125,
    "healthChecks": [
        {
            "gracePeriodSeconds": 100,
            "ignoreHttp1Xx": false,
            "intervalSeconds": 120,
            "maxConsecutiveFailures": 10,
            "path": "/",
            "port": 80,
            "protocol": "HTTP",
            "timeoutSeconds": 300
        }
    ],
    "id": "99026987",
    "instances": 1,
    "mem": 256.0,
    "ports": [80, 8081],
    "requirePorts": true,
    "upgradeStrategy": {
        "maximumOverCapacity": 0.5,
        "minimumHealthCapacity": 0.5
    },
    "uris": [
        "file:///etc/.dockercfg"
    ]
}

error:

{
  "message": "Object is not valid",
  "details": [
    {
      "path": "/healthChecks(0)",
      "errors": [
        "Health check port indices must address an element of the ports array or container port mappings."
      ]
    }
  ]
}
````
ps: using portDefinitions get the same result

marathon 1.1.1
mesos 0.28.1

**Comment by unterstein** [comment]
Hi @max8899,

as far as I can see, the current validation only validates against the port indices, see https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/state/AppDefinition.scala#L758. This validation should also include the port definition.

As quick fix for your issue, try:
```
...
    "healthChecks": [
        {
            "gracePeriodSeconds": 100,
            "ignoreHttp1Xx": false,
            "intervalSeconds": 120,
            "maxConsecutiveFailures": 10,
            "path": "/",
            "portIndex": 0,
            "protocol": "HTTP",
            "timeoutSeconds": 300
        }
    ]
...
```

Kind regards
Johannes

**Comment by unterstein** [comment]
Hi @max8899,

tried to reproduce this with the current marathon 1.3.0-RC6 with your app definition and it validates. Is it possible for you to retest your issue with the current RC? We will ship a final 1.3 release soon.

Kind regards
Johannes

**Comment by max8899** [comment]
Hi @unterstein 

Yes, after searching the source code, I found it's fixed in this issue #3981

**Comment by unterstein** [comment]
🎉
